Source/items.cpp

```diff
@@ -1138,74 +1138,60 @@ void GetBookSpell(int i, int lvl)
 		item[i]._iCurs = ICURS_BOOK_GREY;
 }
 
-void GetStaffPower(int i, int lvl, int bs, unsigned char onlygood)
+void GetStaffPower(int i, int lvl, int bs, BOOL onlygood)
 {
-	int v4;         // esi
-	int v5;         // ebx
-	int v6;         // edx
-	int v7;         // ecx
-	int v9;         // edi
-	int v10;        // ecx
-	int v11;        // ST14_4
-	int v12;        // esi
-	char *v13;      // edi
-	int l[256];     // [esp+Ch] [ebp-484h]
-	char istr[128]; // [esp+40Ch] [ebp-84h]
-	int ia;         // [esp+48Ch] [ebp-4h]
-	char *v17;      // [esp+49Ch] [ebp+Ch]
-
-	v4 = lvl;
-	ia = i;
-	v5 = -1;
+	int l[256];
+	char istr[128];
+	int nl, j, preidx;
+	BYTE addok;
+
+	preidx = -1;
 	if (!random(15, 10) || onlygood) {
-		v6 = 0;
-		v7 = 0;
-		if (PL_Prefix[0].PLPower != -1) {
-			do {
-				if (PL_Prefix[v7].PLIType & 0x100 && PL_Prefix[v7].PLMinLvl <= v4 && (!onlygood || PL_Prefix[v7].PLOk)) {
-					l[v6++] = v7;
-					if (PL_Prefix[v7].PLDouble)
-						l[v6++] = v7;
+		nl = 0;
+		for (j = 0; PL_Prefix[j].PLPower != -1; j++) {
+			if (PL_Prefix[j].PLIType & 0x100 && PL_Prefix[j].PLMinLvl <= lvl) {

```

Comment:
```suggestion
			if (PL_Prefix[j].PLIType & PLT_STAFF && PL_Prefix[j].PLMinLvl <= lvl) {
```
Also not sure if writing `!= 0` for bitmask checks good thing or bad?

Reply:
We don't usually do it

Reply:
When using bitmasks, we are testing true/false i.e. `[!](i & x)` so in that regard it isn't used.